skt/runner.py

```diff
@@ -267,10 +267,18 @@ def cancel_pending_jobs(self):
         """
         Cancel all recipe sets from self.watchlist and remove their IDs from
         self.job_to_recipe_set_map.
+        Cancelling a part of a job leads to cancelling the entire job.
+        So we cancel a job if any of its recipesets is in the watchlist.
         """
         sets_to_cancel = [recipe_set for recipe_set in self.watchlist.copy()]
         if sets_to_cancel:
-            ret = subprocess.call(['bkr', 'job-cancel'] + sets_to_cancel)
+            jobs2cancel = set()
+            for job_id in self.job_to_recipe_set_map:
+                for recipeset in sets_to_cancel:
+                    if recipeset in self.job_to_recipe_set_map[job_id]:
+                        jobs2cancel.add(job_id)
```

Comment:
@veruu I like. That is a nice way to simplify it. But it's slightly less intuitive. From recent experience I know some people would find it confusing; you're expecting someone to know what set intersection is. Don't. :-D

@major Can I get your opinion on this? Do we really want to nitpick every little thing? I don't find it terribly time-effective.
The rule of thumb I use when I review any code is this: I ask myself "does this help us achieve our goals (better)?". If the answer is yes (really yes), I don't nitpick much. If the answer is no (for any reason), then I request changes.

Reply:
Hey, it was just an alternative idea to think about (that's why I commented instead of pushing the "request changes" button)! It's faster, shorter, uses python's builtin functional programming and TBH (if you don't count our lengthy variable names) I find it cleaner then the nested for loops over different fields which I had to draw to figure out. And yes, I expect people to check the docs if they don't know the methods used (and comments exist too, if needed).

I always try to find things that might help with eg. future maintainability or shortening the code. Please don't take it personally, because it's not meant that way at all. We had the discussion about some code that does the job, but is a maintenance hell :) so when reviewing, instead of "fixes the bug, good enough to merge" mentality, I go with the "can the same thing be done better?", where "better" can be almost anything - shorter, faster, cleaner, using a builtin functionality instead of implementing your own wheel etc. It doesn't save time if someone has to come later and rewrite it because our implementation doesn't handle some edge cases a builtin or a module does (see kpet and patch parsing bugs), for example. You get the idea, I hope.

I'm open to discussing things and if the team agrees that a "good enough" approach without thinking about alternative solutions is the way to, we can get a webhook to automerge pulls that pass the Travis and CircleCI.

Reply:
@veruu Aah, right, you didn't click RFC button. Sorry about that – it's been a while since I've used github. 

Anyway, I'll respin - travis-ci is stuck anyway, so I'll need to push again.
I kindof agree with your logic. But even so, don't expect it to be necessarily more readable for (some) other people.